### PR TITLE
Fork Analysis

### PR DESCRIPTION
**Pull request by praetoriansentry**
# Description

Creating a script to take a forked block hash and work it's way back the chain until it reaches a canonical block

# Testing

I ran this against `39599624` from mainnet and was able to export blocks from side chain and canonical as expected.

### CODE REVIEW

File: cmd/fork/fork.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+)
+
+var ForkCmd = &cobra.Command{
+	Use:   "fork blockhash http://polygon-rpc.com",
+	Short: "Take a forked block and walk up the chain to do analysis",
+	Long: `
+TODO
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		log.Info().Msg("Hi there")
```
Comment (leovct) [inline cmd/fork/fork.go:35]: you might want to remove this debug line :)

**Review by leovct — APPROVED** [review]
lgtm

File: cmd/fork/fork.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+var (
+	rpcURL     string
+	blockHash  ethcommon.Hash
+	retryLimit = 30
+
+	errRetryLimitExceeded = fmt.Errorf("Unable to process request after hitting retry limit")
+)
+
+var ForkCmd = &cobra.Command{
+	Use:   "fork blockhash http://polygon-rpc.com",
```
Comment (leovct) [inline cmd/fork/fork.go:29]: `https://polygon-rpc.com` with `https` instead of `http`

**Review by gatsbyz — APPROVED** [review]
lgtm

**Pull request merged**